src/docfx/lib/schema/JsonSchemaTransform.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class JsonSchemaTransform
+    {
+        public static (List<Error> errors, JToken token) Transform(Document file, Context context, JsonSchema schema, JToken token, Action<Document> buildChild)
+        {
+            var errors = new List<Error>();
+
+            Transform(file, context, schema, token, errors, buildChild);
+            return (errors, token);
+        }
+
+        private static void Transform(Document file, Context context, JsonSchema schema, JToken token, List<Error> errors, Action<Document> buildChild)
+        {
+            switch (token)
+            {
+                case JValue scalar:
+                    token.Replace(TransformScalar(file, context, schema, scalar, errors, buildChild));
```

Comment:
this will just replace the value and keep the line info? can we do this in other places as well?

Reply:
TransformScalar return an new JToken with line info, so line info will be there